spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/configuration/TaskBatchAutoConfiguration.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.task.batch.configuration;
+
+import javax.sql.DataSource;
+
+import org.springframework.batch.core.Job;
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.cloud.task.batch.listener.TaskBatchDao;
+import org.springframework.cloud.task.batch.listener.TaskBatchExecutionListener;
+import org.springframework.cloud.task.batch.listener.support.JdbcTaskBatchDao;
+import org.springframework.cloud.task.batch.listener.support.MapTaskBatchDaoFactoryBean;
+import org.springframework.cloud.task.configuration.EnableTask;
+import org.springframework.cloud.task.repository.support.SimpleTaskExplorer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * Provides auto configuration for the {@link TaskBatchExecutionListener}.
+ *
+ * @author Michael Minella
+ */
+@Configuration
+@ConditionalOnClass({Job.class, EnableTask.class})
+public class TaskBatchAutoConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean
+	public TaskBatchExecutionListenerBeanPostProcessor batchTaskExecutionListenerBeanPostProcessor() {
+		return new TaskBatchExecutionListenerBeanPostProcessor();
+	}
+
+	@Configuration
+	@ConditionalOnBean(DataSource.class)
+	@ConditionalOnMissingBean(name = "batchTaskExecutionListener")
+	public static class JdbcTaskBatchExecutionListenerConfiguration {
+
+		@Bean
+		public TaskBatchExecutionListener batchTaskExecutionListener(DataSource dataSource) {
+			return new TaskBatchExecutionListener(new JdbcTaskBatchDao(dataSource));
+		}
+	}
+
+	@Configuration
+	@Conditional(NoDataSourceCondition.class)
+	@ConditionalOnMissingBean(name = "batchTaskExecutionListener")
+	public static class MapTaskBatchExecutionListenerConfiguration {
+
+		@Bean
+		@ConditionalOnMissingBean
+		public TaskBatchExecutionListener batchTaskExecutionListener(SimpleTaskExplorer taskExplorer) throws Exception {
+			return new TaskBatchExecutionListener(taskBatchDao(taskExplorer));
+		}
+
+		@Bean
+		@ConditionalOnMissingBean
+		public TaskBatchDao taskBatchDao(SimpleTaskExplorer taskExplorer) throws Exception {
```

Comment:
By doing lines 75 and 76, Spring will replace that method call at line 72 with injecting the result of this method.  One thing I should change is I should be returning the FactoryBean and letting Spring handle the lifecycle instead of managing it myself.